src/nvim/api/vim.c

```diff
@@ -107,11 +108,17 @@ String nvim_exec(String src, Boolean output, Error *err)
   if (output) {
     msg_silent++;
   }
+
+  const sctx_T save_current_sctx = current_sctx;
+  set_api_sctx(channel_id);
```

Comment:
if set_api_sctx returns the old current_sctx, could it help signal that something is supposed to be done with it? also saves a line of code.

```suggestion
  const sctx_T save_current_sctx = set_api_sctx(channel_id);
```

Reply:
done